### PR TITLE
Update ADC test, add WZR/WSP, ADCS, SBCS test

### PR DESCRIPTION
**Pull request by MS-DOS1999**

### CODE REVIEW

File: Ryujinx.Tests/Cpu/CpuTestAlu.cs [hunk 1]
```diff
@@ -5,13 +5,34 @@ namespace Ryujinx.Tests.Cpu
 {
     public class CpuTestAlu : CpuTest
     {
-        [TestCase(2u, 3u, 6ul, true)]
-        [TestCase(2u, 3u, 5ul, false)]
-        public void Adc(uint A, uint B, ulong Result, bool CarryTest)
+        public void Adc()
```
Comment (LDj3SNuD) [inline Ryujinx.Tests/Cpu/CpuTestAlu.cs]: [Test] or [TestCase]

File: Ryujinx.Tests/Cpu/CpuTestAlu.cs [hunk 2]
```diff
@@ -5,13 +5,34 @@ namespace Ryujinx.Tests.Cpu
...
+            Reset();
+
+            ThreadState = SingleOpcode(0x9A020020, X1: 2, X2: 3, Carry: false);
+            Assert.AreEqual(5, ThreadState.X0);
+
+            Reset();
+
+            // ADC W0, W1, W2
+            ThreadState = SingleOpcode(0x1A020020, X1: 2, X2: 3, Carry: true);
+            Assert.AreEqual(6, ThreadState.X0);
```
Comment (LDj3SNuD) [inline Ryujinx.Tests/Cpu/CpuTestAlu.cs]: If before we have a failed test this "new" opcode is not tested: use TestCase (s); or divide the test into two tests (Xs; Ws); or use Assert.Multiple(() => {});

File: Ryujinx.Tests/Cpu/CpuTestAlu.cs [hunk 3]
```diff
@@ -5,13 +5,34 @@ namespace Ryujinx.Tests.Cpu
...
+
+            Reset();
+
+            ThreadState = SingleOpcode(0x1A020020, X1: 2, X2: 3, Carry: false);
+            Assert.AreEqual(5, ThreadState.X0);
+
+            Reset();
+
+            // ADC Overflow
+            ThreadState = SingleOpcode(0x1A020020, X1: 0xFFFFFFFF, X2: 0x2, Carry: false);
```
Comment (LDj3SNuD) [inline Ryujinx.Tests/Cpu/CpuTestAlu.cs]: ADC does not update the status of the flags -> test always failed
ADCS does it, but it has not been implemented yet

**Comment by gdkchan** [comment]
fyi @MerryMage

File: Ryujinx.Tests/Cpu/CpuTestAlu.cs [hunk 4]
```diff
@@ -22,6 +43,36 @@ public void Add()
...
+        [TestCase(5u)]
+        [TestCase(7u)]
+        [TestCase(0xFFFFFFFFu)]
+        [TestCase(0xFFFFFFFBu)]
+        public void Adds(uint A)
+        {
+            //ADDS WZR, WSP, #5
+            AThreadState ThreadState = SingleOpcode(0x310017FF, X31: A);
+            Assert.IsFalse(ThreadState.Negative);
+            if(A == 0xFFFFFFFB)
```
Comment (MerryMage) [inline Ryujinx.Tests/Cpu/CpuTestAlu.cs]: Consider instead:
```
        [TestCase(2u,          false, false)]
        [TestCase(5u,          false, false)]
        [TestCase(7u,          false, false)]
        [TestCase(0xFFFFFFFFu, false, true )]
        [TestCase(0xFFFFFFFBu, true,  true )]
        public void Adds(uint A, bool Zero, bool Carry)
        {
            //ADDS WZR, WSP, #5
            AThreadState ThreadState = SingleOpcode(0x310017FF, X31: A);
            Assert.IsFalse(ThreadState.Negative);
            Assert.AreEqual(Zero, ThreadState.Zero);
            Assert.AreEqual(Carry, ThreadState.Carry);
            Assert.IsFalse(ThreadState.Overflow);
            Assert.AreEqual(A, ThreadState.X31);
        }
```

When using test cases ideally data regarding the test case should be within the test case.

**Comment by gdkchan** [comment]
lgtm, thank you!

**Pull request merged**